pkg/kubelet/prober/prober_test.go

```diff
@@ -333,3 +335,141 @@ func TestNewExecInContainer(t *testing.T) {
 		}
 	}
 }
+
+func TestNewProber(t *testing.T) {
+	runner := &containertest.FakeContainerCommandRunner{}
+	recorder := &record.FakeRecorder{}
+	prober := newProber(runner, recorder)
+
+	tests := []struct {
+		name   string
+		got    interface{}
+		expect interface{}
+	}{
+		{"non-nil prober", prober != nil, true},
+		{"runner set", prober.runner, runner},
+		{"recorder set", prober.recorder, recorder},
+		{"exec probe initialized", prober.exec != nil, true},
+		{"http probe initialized", prober.http != nil, true},
+		{"tcp probe initialized", prober.tcp != nil, true},
+		{"grpc probe initialized", prober.grpc != nil, true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.got != tt.expect {
+				t.Errorf("Expected %v, but got %v", tt.expect, tt.got)
+			}
+		})
+	}
+}
+
+func TestExecInContainer_Start(t *testing.T) {
+	ctx := context.Background()
+	runner := &containertest.FakeContainerCommandRunner{
+		Stdout: "executed",
+	}
+	prober := &prober{
+		runner: runner,
+	}
+
+	container := v1.Container{}
+	containerID := kubecontainer.ContainerID{Type: "docker", ID: "containerID"}
+	cmd := []string{"ls", "-l"}
+
+	exec := prober.newExecInContainer(ctx, container, containerID, cmd, 0)
+	err := exec.Start()
+	if err != nil {
+		t.Errorf("Expected no error, got %v", err)
+	}
```

Comment:
please, use `require.Noerror(err)` or `assert.NoError(err)` in this and similar cases.